core/src/main/java/org/elasticsearch/common/compress/Compressor.java

```diff
@@ -31,5 +32,16 @@
 
     StreamInput streamInput(StreamInput in) throws IOException;
 
+    /**
+     * Creates a new stream output that compresses the contents and writes to the provided stream
+     * output. Closing the returned {@link StreamOutput} will close the provided stream output.
+     */
     StreamOutput streamOutput(StreamOutput out) throws IOException;
+
+    /**
+     * Creates a new stream output that compresses the contents and writes to the provided stream
+     * output. Closing the returned {@link StreamOutput} will not close the provided stream output
+     * as closing would also release the bytes of the {@link ReleasableBytesStreamOutput}
+     */
+    StreamOutput streamOutput(ReleasableBytesStreamOutput out) throws IOException;
```

Comment:
now that I am thinking about it I wonder if we should do this at all and just let the caller wrap the stream in a non-closeable stream. I'd prefer this over such an overloaded method. it's not clear when this one applies from the interface perspective.